indico/modules/categories/controllers/display.py

```diff
@@ -591,22 +594,41 @@ def _process(self):
                           .options(load_only('id', 'title', 'start_dt', 'end_dt', 'timezone'))
                           .order_by(Event.title)
                           .all())
-        return jsonify_data(flash=False, events=events, ongoing_event_count=len(ongoing_events),
+        return jsonify_data(flash=False, events=events, categories=categories, locations=locations,
+                            group_by=self.group_by.name,
+                            ongoing_event_count=len(ongoing_events),
                             ongoing_events_html=self._render_ongoing_events(ongoing_events))
 
+    def _find_nearest_category(self, category_chain):
+        for index, category_data in enumerate(category_chain):
+            if category_data['id'] == self.category.id:
+                if index == len(category_chain) - 1:
+                    return category_data
+                else:
+                    return category_chain[index + 1]
+        # this should never happen
+        raise NotFound
```

Comment:
`NotFound` is an HTTP error, it will be silent to instance admins (since it's just a 404 for the client). Better use a plain `Exception` for it (since that'll be logged as an error)

Reply:
Yep, that sounds good.